util/lang-compress.pl

```diff
@@ -0,0 +1,189 @@
+#! /usr/bin/env perl
+#
+# C source compressor.  This:
+#
+# - merges continuation lines
+# - removes comments (not in strings)
+# - removes empty lines (not in strings)
+
+use strict;
+use warnings;
+
+my $debug = defined $ENV{DEBUG};
+my $lang = shift @ARGV;
```

Comment:
Can't you really intuit the `$lang` from the extension? OpenSSL is pretty consistent even if the world isn't :)

Reply:
I prefer to have that done on demand, i.e. let the caller do the detection.

Reply:
Part of it is that it's designed to be part of a pipe...  you know, like many little unix tools :wink: